### DISCUSSION TITLE
Xiaomi Smart Air Purifier Elite

### DISCUSSION THREAD

**Discussion by Ostromogilski** [merdok/homebridge-miot#489] [General]
Please add support of Xiaomi Smart Air Purifier Elite (Model: zhimi.airp.meb1)

**Comment by Ostromogilski** [comment]
here are available properties for device
```Device properties: [
  "air-purifier:on",
  "air-purifier:fault",
  "air-purifier:mode",
  "air-purifier:fan-level",
  "air-purifier:plasma",
  "air-purifier:uv",
  "environment:relative-humidity",
  "environment:pm2.5-density",
  "environment:temperature",
  "environment:pm10-density",
  "environment:air-quality",
  "filter:filter-life-level",
  "filter:filter-used-time",
  "alarm:alarm",
  "physical-controls-locked:physical-controls-locked",
  "screen:brightness",
  "air-purifier-favorite:fan-level",
  "device-display-unit:temperature-display-unit",
  "custom-service:moto-speed-rpm",
  "custom-service:reboot-cause",
  "custom-service:iic-error-count",
  "custom-service:country-code",
  "custom-service:favorite-square",
  "filter-time:filter-used-debug",
  "aqi:aqi-updata-heartbeat",
  "rfid:rfid-tag",
  "rfid:rfid-factory-id",
  "rfid:rfid-product-id",
  "rfid:rfid-time",
  "rfid:rfid-serial-num"
]

**Comment by merdok** [comment]
Hi,
does auto discovery not work for the device? Did you already to add it to the plugin?

**Comment by Ostromogilski** [comment]
Hi! Yes, main functionality works with auto discovery.
But `UV` and `Plasma` switches are absent.
@merdok Please let me know if I can be of help by providing the logs etc.

**Comment by Ostromogilski** [comment]
I was able to add `Plasma` and `UV` switches
``` "propertyControl": [
                {
                    "property": "air-purifier:plasma",
                    "name": "Plasma"
                },
                {
                    "property": "air-purifier:uv",
                    "name": "UV"
                }

**Comment by merdok** [comment]
Yes, this is the correct way to add functionality which is missing. The devices get more and more new features so implementing them one by one would take quite a lot of work, hence i added the possibility to expose manually whatever properties are needed 😊
Is now everything working what you expect?

**Comment by Ostromogilski** [comment]
Works like a charm! Thank you @merdok

**Comment by WildEchoWanderer** [comment]
I bought this device and now i am wondering if it works with the "Xiaomi Miot Auto" Integration or if i need to add it with HACS or even Custom?

**Reply by merdok** [reply]
You can try to add and you will see what will happen. Should be discovered properly I think.

**Reply by WildEchoWanderer** [reply]
Adding worked perfectly as it should. 
But it only imports the values (PM2.5, PM10, Temp, Humidity, Filter life, Filter use time & air quality), Lock physical controll & a ON/OFF button.
I am missing the Mode (A / Night / Manual), the strenght in manual, Plasma & UV ON/OFF. Also the display illumination is missing. How can i add this? (Saw in this tread i can add a line of code, but where?)

**Reply by merdok** [reply]
You have to use propertyControl to expose the stuff what is missing. Have a look at the README.

**Reply by WildEchoWanderer** [reply]
i now realised this is for homebridge and not homeassistant...
